crates/analyzer-host/src/lsp_impl/active_initialized.rs

```diff
@@ -172,6 +173,7 @@ async fn on_text_document_did_change(
 	}
 
 	// TODO: avoid cloning
+	// implement LspPos::lazy_add
```

Comment:
Currently lazy_add is never used but this is where it would go. `analyzer.update()` is where the LspPos gets updated but that will bypass the use of `TextDocumentContentChangeEvent`. Wondering if the best way would be for `update()` to take a `Buffer` argument instead of file?